pkg/framework/context/context.go

```diff
@@ -224,19 +233,29 @@ func (p *Provider) InboundMessageHandler() transport.InboundMessageHandler {
 }
 
 func (p *Provider) getDIDs(envelope *transport.Envelope) (string, string, error) {
-	myDID, err := p.didConnectionStore.GetDID(base58.Encode(envelope.ToKey))
-	if errors.Is(err, did.ErrNotFound) {
-	} else if err != nil {
-		return "", "", fmt.Errorf("failed to get my did: %w", err)
-	}
+	var (
+		myDID    string
+		theirDID string
+		err      error
+	)
+
+	return myDID, theirDID, backoff.Retry(func() error {
+		var notFound bool
+		myDID, err = p.didConnectionStore.GetDID(base58.Encode(envelope.ToKey))
```

Comment:
@soluchok can you please add a comment/document on why we need the retry ? The function fetches the mapping between the key and DID from the store (this is not actual DID Resolution).

Reply:
When a connection exists for `envelope.ToKey` then it should exist for `envelope.FromKey`.
The bug I found was bc we had a connection for `envelope.ToKey` but we didn't for   `envelope.FromKey`.
Now I added `notFound` flag which allows us to determine whether the connection for `envelope.FromKey` should be present. If it should be present but the actual result is not found we will do a retry logic.